### PR TITLE
new profile data methods

### PR DESCRIPTION
**Pull request by owocki**
##### Description

this commit adds methods for calculating all the info needed on profiles to give them activity data

##### Refers/Fixes

https://github.com/gitcoinco/web/issues/4908

##### Testing

locally

https://gitcoincore.slack.com/archives/CMGLVLSJX/p1568078390010700g

### CODE REVIEW

**Comment by iamonuwa** [comment]
Super dope

File: app/dashboard/models.py [hunk 1]
```diff
@@ -2105,6 +2129,12 @@ class Profile(SuperModel):
         help_text=_('override profile avatar'),
     )
     dominant_persona = models.CharField(max_length=25, choices=PERSONAS, blank=True)
+    longest_streak = models.IntegerField(default=0)
+    activity_level = models.CharField(max_length=10, blank=True, help_text=_('the users activity level (high, low, new)'))
```
Comment (frankchen07) [inline app/dashboard/models.py]: did you forget `medium`? (I guess it's only the help text)

File: app/dashboard/models.py [hunk 2]
```diff
@@ -2400,6 +2430,136 @@ def is_staff(self):
...
+            return "New"
+
+        visits = self.actions.filter(action='Visit')
+        visits_last_month = visits.filter(created_on__gt=timezone.now() - timezone.timedelta(days=30)).count()
+
+        if visits_last_month > high_threshold:
+            return "High"
+        if visits_last_month > med_threshold:
+            return "Medium"
+        if visits_last_month > med_threshold:
```
Comment (frankchen07) [inline app/dashboard/models.py]: should this be `visits_last_month` < `med_threshold`?

File: app/dashboard/models.py [hunk 3]
```diff
@@ -2400,6 +2430,136 @@ def is_staff(self):
...
+
+        if self.created_on > (timezone.now() - timezone.timedelta(days=new_threshold_days)):
+            return "New"
+
+        visits = self.actions.filter(action='Visit')
+        visits_last_month = visits.filter(created_on__gt=timezone.now() - timezone.timedelta(days=30)).count()
+
+        if visits_last_month > high_threshold:
+            return "High"
+        if visits_last_month > med_threshold:
```
Comment (frankchen07) [inline app/dashboard/models.py]: `med_threshold` is the same as `high_threshold`, does that make sense? also - should there be an upper bound as well? (`visits_last_month` > `med_threshold` and <= `high_threshold`?)

File: app/dashboard/models.py [hunk 4]
```diff
@@ -2400,6 +2430,136 @@ def is_staff(self):
...
+        """
+        bounties = self.bounties
+        completed_bounties = bounties.filter(idx_status__in=['done'])
+        relationships = []
+        for bounty in completed_bounties:
+            fulfiller_handles = bounty.fulfillers_handles
+            if bounty.is_funder(self.handle):
+                for handle in fulfiller_handles:
+                    relationships.append(handle)
+            if self.handle in fulfiller_handles:
```
Comment (frankchen07) [inline app/dashboard/models.py]: nice this handles both funder -> fulfiller and and vice versa

File: app/dashboard/models.py [hunk 5]
```diff
@@ -2400,6 +2430,136 @@ def is_staff(self):
...
+        Returns:
+            int; the success percentage for this users bounties as a positive integer.
+
+        """
+        bounties = self.bounties
+        completed_bounties = bounties.filter(idx_status__in=['done'])
+        eligible_bounties = bounties.filter(idx_status__in=['done', 'expired', 'cancelled'])
+
+        if eligible_bounties.count() == 0:
+            return -1
```
Comment (frankchen07) [inline app/dashboard/models.py]: hm why -1?

File: app/dashboard/models.py [hunk 6]
```diff
@@ -2400,6 +2430,136 @@ def is_staff(self):
...
+        iterdate = start_date
+        max_streak = 0
+        this_streak = 0
+        while iterdate < end_date:
+            # housekeeping
+            last_iterdate = start_date
+            iterdate += timezone.timedelta(days=1)
+
+            is_weekday = iterdate.weekday() < 5
+            if not is_weekday:
```
Comment (frankchen07) [inline app/dashboard/models.py]: shouldn't weekends be counted in the streak as well? 

also is the `if not is_weekday` grabbing weekends here? if its 5 or greater `is_weekday` is false and the `if not` will render `true`, and it'll continue, no?

Comment (owocki) [reply app/dashboard/models.py]: derp it should only count weekdays..  will update

 i dont think weekends should count in the streak mechanic.. we dont want to build a culture that encourages burnout

File: app/dashboard/models.py (hunk 4 above)
Comment (owocki) [reply app/dashboard/models.py]: yar

File: app/dashboard/models.py (hunk 5 above)
Comment (owocki) [reply app/dashboard/models.py]: its a way of saying "this user hasnt done any bounties; pls dont display on frontend". as opposed to 0 which is for when the user has made $0/hr for many bounties

File: app/dashboard/models.py (hunk 1 above)
Comment (owocki) [reply app/dashboard/models.py]: in the help text only yes

File: app/dashboard/models.py (hunk 3 above)
Comment (owocki) [reply app/dashboard/models.py]: yeah.... it is.  

alos i just made this up... you guys tell me what the thresholds should be pls. in particular, @frankchen07 curious your opionion

File: app/dashboard/models.py (hunk 2 above)
Comment (owocki) [reply app/dashboard/models.py]: no; d1574cc644def2763c9e6435530456028a16eb1c

File: app/dashboard/models.py (hunk 3 above)
Comment (frankchen07) [reply app/dashboard/models.py]: I think it's fine where it is 15 stands to be a visit every other day, medium is 2 visit per week, and lower than that is low.

**Pull request merged**